tests/test_date.py

```diff
@@ -118,163 +192,230 @@ def test_should_one_date_for_each_month(self, low, high, expected_results):
+        self.given_now(datetime(2014, 8, 12))
+        self.given_utc_date()
+        self.when_date_is_parsed_with_format('August 2014', '%B %Y')
+        self.then_dates_is_not_none()
+        self.then_period_equal('month')
+        self.then_dates_are_equal()
+
+    def given_day(self, day):
+        self.day = day
+
+    def given_month(self, month):
+        self.month = month
+
+    def given_invalid_date(self, date_str):
+        self.invalid_date = date_str
 
-        result = date.parse_with_formats('09.16', ["%m.%d"])
-        self.assertEquals(datetime(today.year, 9, 16).date(), result['date_obj'].date())
+    def given_current_year(self):
+        self.year = datetime.today().year
 
-    def test_should_use_current_date_for_dates_without_day(self):
-        twelfth = datetime(2014, 8, 12)
-        datetime_mock = Mock(wraps=datetime)
-        datetime_mock.utcnow = Mock(return_value=twelfth)
+    def given_a_date(self, date_obj):
+        self.date_obj = date_obj.date()
 
-        with patch('dateparser.date_parser.datetime', new=datetime_mock):
-            dt_data = date.parse_with_formats('August 2014', ['%B %Y'])
+    def given_now(self, given_date):
+        self.datetime_mock = Mock(wraps=datetime)
+        self.datetime_mock.utcnow = Mock(return_value=given_date)
 
-        self.assertIsNotNone(dt_data)
-        self.assertEquals('month', dt_data['period'])
-        self.assertEquals(datetime(2014, 8, 31).date(), dt_data['date_obj'].date())
+    def given_utc_date(self):
+        self.add_patch(patch('dateparser.date_parser.datetime', new=self.datetime_mock))
 
-        twelfth = datetime(2014, 2, 12)
-        datetime_mock = Mock(wraps=datetime)
-        datetime_mock.utcnow = Mock(return_value=twelfth)
+    def when_date_is_parsed_with_format(self, some_date, some_format):
+        self.dt_data = date.parse_with_formats(some_date, [some_format])
 
-        with patch('dateparser.date_parser.datetime', new=datetime_mock):
-            dt_data = date.parse_with_formats('February 2014', ['%B %Y'])
+    def then_should_use_current_year_for_dates_without_year(self):
+        self.assertEquals(datetime(self.year, self.month, self.day).date(), self.dt_data['date_obj'].date())
 
-        self.assertIsNotNone(dt_data)
-        self.assertEquals('month', dt_data['period'])
-        self.assertEquals(datetime(2014, 2, 28).date(), dt_data['date_obj'].date())
+    def then_parsed_date_is(self):
+        self.assertEquals(self.date_obj, self.dt_data['date_obj'].date())
 
+    def then_shouldnt_parse_invalid_date(self):
+        self.parsed_invalid_date = date.parse_with_formats(self.invalid_date,
+                                                           ['%Y-%m-%d'])['date_obj']
+        self.assertIsNone(self.parsed_invalid_date)
 
-class DateDataParserTest(unittest.TestCase):
+    def then_dates_is_not_none(self):
+        self.assertIsNotNone(self.dt_data)
 
+    def then_period_equal(self, month):
```

Comment:
I had it like that because that is what I was passing in. Refactored to use ```period``` instead.